c/src/core/transport.c

```diff
@@ -2828,6 +2861,8 @@ uint32_t pn_transport_get_max_frame(pn_transport_t *transport)
 void pn_transport_set_max_frame(pn_transport_t *transport, uint32_t size)
 {
   // if size == 0, no advertised limit to input frame size.
+  if (transport->open_sent)
+    return;  // Too late.  Silently disregard request.
```

Comment:
Instead of silently disregarding the request it's probably better to log a warning.